src/librustdoc/html/templates/page.html

```diff
@@ -8,7 +8,7 @@
     <title>{{page.title}}</title> {# #}
     <script>if(window.location.protocol!=="file:") {# Hack to skip preloading fonts locally - see #98769 #}
     for(f of "{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",")) {# #}
-     document.write(`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`) {# #}
+     document.head.innerHTML += `<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">` {# #}
```

Comment:
innerHTML is not great either since it manipulates the string representation of the HTML, which means it removes all the elements and then reinserts them. Which probably has weird effects when that happens during processing of the `head` itself.

The modern way to do things is to use `insertAdjacentHTML` which parses the string as a `DocumentFragment` and then appends it to the child list.

```javascript
document.head.insertAdjacentHTML("beforeend", `<link rel="preload" as="font" type="font/woff2" crossorigin ref="{{static_root_path|safe}}${f}">`)
```

Reply:
I mean we can even use `createElement` and `appendChild` to avoid the string parsing altogether.

Reply:
But then it would increase all the files size again. ^^'

However we can do better: having only one `insertAdjacentHTML`.